### PR TITLE
ENG-13856, fix various issues found in system tests.

### PR DESCRIPTION
**Pull request by luyangco**
fix thread-safe issue when we touching transaction backlog,
fix a corner case that fragment message doesn't get cleaned by restart completion message,
fix an issue that we don't flush backlog for certain non-restartable MP sysprocs (e.g. @SwapTablesCore)

Change-Id: I39419514a474bc837c3f9e0f61f8317c5b912fb7

### CODE REVIEW

File: src/frontend/org/voltdb/iv2/Scoreboard.java [hunk 1]
```diff
@@ -55,18 +55,19 @@ public void addCompletedTransactionTask(CompleteTransactionTask task, Boolean mi
...
-        // special case, scoreboard is empty
-        if (m_compTasks.peekFirst() == null) {
-            m_compTasks.addFirst(Pair.of(task, missingTxn));
-            return;
-        }
-
         // Restart completion steps on any pending prior fragment of the same transaction
         if (task.getTimestamp() != CompleteTransactionMessage.INITIAL_TIMESTAMP &&
                 MpRestartSequenceGenerator.isForRestart(task.getTimestamp()) &&
                 m_fragTask != null && m_fragTask.getTxnId() == task.getMsgTxnId()) {
```
[the remark this answers is not in the document]
Comment (wweiss-voltdb) [reply src/frontend/org/voltdb/iv2/Scoreboard.java]: I think it needs to stay. It is possible to receive restarts for fragments that have already been processed and therefore m_fragTask would be null and the txnId comparison would generate an NPE
